growattServer/__init__.py

```diff
@@ -128,106 +132,439 @@ def login(self, username, password, is_password_hashed=False):
 
     def inverter_data(self, inverter_id, date=None):
         """
         Get inverter data for specified date or today.
+
+        Args:
+            inverter_id (str): The ID of the inverter.
+            date (datetime, optional): The date you are interested in. Defaults to datetime.datetime.now().
+
+        Returns:
+            dict: A dictionary containing the inverter data.
+
+        Raises:
+            Exception: If the request to the server fails.
         """
         date_str = self.__get_date_string(date=date)
-        response = self.session.get(self.get_url('newInverterAPI.do'), params={
+        response = self.session.get(self.__get_url('newInverterAPI.do'), params={
             'op': 'getInverterData',
             'id': inverter_id,
             'type': 1,
             'date': date_str
         })
-        data = json.loads(response.content.decode('utf-8'))
-        return data
+
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise Exception('Failed to retrieve inverter data')
 
     def inverter_detail(self, inverter_id):
         """
-        Get "All parameters" from PV inverter.
+        Get detailed data from PV inverter.
+
+        Args:
+            inverter_id (str): The ID of the inverter.
+
+        Returns:
+            dict: A dictionary containing the inverter details.
+
+        Raises:
+            Exception: If the request to the server fails.
         """
-        response = self.session.get(self.get_url('newInverterAPI.do'), params={
+        response = self.session.get(self.__get_url('newInverterAPI.do'), params={
             'op': 'getInverterDetailData',
             'inverterId': inverter_id
         })
 
-        data = json.loads(response.content.decode('utf-8'))
-        return data
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise Exception('Failed to retrieve inverter details')
 
     def inverter_detail_two(self, inverter_id):
         """
-        Get "All parameters" from PV inverter.
+        Get detailed data from PV inverter (alternative endpoint).
+
+        Args:
+            inverter_id (str): The ID of the inverter.
+
+        Returns:
+            dict: A dictionary containing the inverter details.
+
+        Raises:
+            Exception: If the request to the server fails.
         """
-        response = self.session.get(self.get_url('newInverterAPI.do'), params={
+        response = self.session.get(self.__get_url('newInverterAPI.do'), params={
             'op': 'getInverterDetailData_two',
             'inverterId': inverter_id
         })
 
-        data = json.loads(response.content.decode('utf-8'))
-        return data
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise Exception('Failed to retrieve inverter details')
+
+    def tlx_get_system_status(self, plant_id, tlx_id):
```

Comment:
```suggestion
    def tlx_system_status(self, tlx_id, plant_id):
```
pulls it inline with 
https://github.com/indykoning/PyPi_GrowattServer/blob/75b6b73d3aa7f4014ce6759a3067d5c1375e74f3/growattServer/__init__.py#L308